hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -806,6 +878,225 @@ public Bytes getFailureResultFor(final ResponseCodeEnum status) {
+	 *     {@link TokenCreateWrapper}.
+	 *     		<ul>
+	 *     		 	<li><b>result</b> - (TBD) the {@link DecodingFacade} throws an exception and null is returned
+	 *     		 	from the {@link HTSPrecompiledContract}</li>
+	 *     		 	<li><b>gas cost</b> - TBD</li>
+	 *     		</ul>
+	 *     	</li>
+	 *     <li>The decoding succeeds, we create a valid {@link TokenCreateWrapper}, but we cannot translate it to a
+	 *     valid token create {@link TransactionBody}. This comes from <b>difference in the design of the Solidity
+	 *     function interface and the HAPI API (protobufs)</b>
+	 *     		<ul>
+  	 *     		 	<li><b>result</b> - {@link MessageFrame}'s revertReason is set to the specific error and null is
+	 *     		 	returned from the {@link HTSPrecompiledContract}</li>
+	 * 	    		<li><b>gas cost</b> - TBD</li>
+	 *     		</ul>
+	 *     	</li>
+	 *     <li>The decoding succeeds, we create a valid {@link TokenCreateWrapper}, we successfully translate it to a
+	 *     valid token create {@link TransactionBody}. However, the {@link CreateChecks} validations find an input
+	 *     error.
+	 *     		<ul>
+	 *     		 	<li><b>result</b> - a child {@link ExpirableTxnRecord} is created, containing the error response
+	 *     		 	code. (from the point of view of the EVM this is a successful precompile call, however, from a
+	 *     		 	Hedera's perspective there has been a problem during the execution)
+	 *     		 	</li>
+	 * 	    		<li><b>gas cost</b> - TBD</li>
+	 *     		</ul>
+	 *     	</li>
+	 *     <li>The decoding succeeds, we create a valid {@link TokenCreateWrapper}, we successfully translate it to a
+	 *     valid token create {@link TransactionBody}, the {@link CreateChecks} token create validations pass and the
+	 *     whole execution flow succeeds.
+	 *     		<ul>
+	 *     		 	<li><b>result</b> - a child {@link ExpirableTxnRecord} is created, containing the successful
+	 *     		 	response code and the ID of the newly created token.</li>
+	 * 	    		<li><b>gas cost</b> - TBD</li>
+	 *     		</ul>
+	 *     	</li>
+	 * </ol>
+	 */
+	protected class TokenCreatePrecompile implements Precompile {
+		protected TokenCreateWrapper tokenCreateOp;
+
+		@Override
+		public TransactionBody.Builder body(Bytes input, UnaryOperator<byte[]> aliasResolver) {
+			tokenCreateOp = switch (functionId) {
+				case ABI_ID_CREATE_FUNGIBLE_TOKEN -> decoder.decodeFungibleCreate(input, aliasResolver);
+				case ABI_ID_CREATE_FUNGIBLE_TOKEN_WITH_FEES ->
+						decoder.decodeFungibleCreateWithFees(input, aliasResolver);
+				case ABI_ID_CREATE_NON_FUNGIBLE_TOKEN -> decoder.decodeNonFungibleCreate(input, aliasResolver);
+				case ABI_ID_CREATE_NON_FUNGIBLE_TOKEN_WITH_FEES ->
+						decoder.decodeNonFungibleCreateWithFees(input, aliasResolver);
+				default -> throw new InvalidTransactionException(String.format(UNKNOWN_FUNCTION_ID_ERROR_MESSAGE,
+						"Create", functionId, input), FAIL_INVALID);
```

Comment:
We should return `null` here and handle this the same way we handle it in `prepareComputation()` -> default to null and handle it in `compute()`.
Generally we want to reserve `FAIL_INVALID` for internal failures in our system, in this case it's actually bad input to the precompile -> user's error.

I noticed the same problem for `TransferPrecompile` and `ERCTransferPrecompile`, will fix it in a separate PR.

Reply:
I think we have added the default cases in all of those precompiles solely to keep Sonar happy.
If during the execution we reach the `body()` method of one of the precompiles doesn't it mean that the switch statement in `prepareComputation()` has found a matching function and has correctly instantiated the corresponding precompile? Otherwise, it would never call `decodeInput()`, which in turn calls  `body()`.